### PR TITLE
Add parameter passing to the CBC interface

### PR DESCRIPTION
**Pull request by yingzong**
Previously, CBCInterface does not read from solver-specific
parameters because there is no ReadParameterFile implemented.

This patch implements a ReadParameterFile that takes these
parameters, saves them in the CBCInterface instance, and passes
them to callCbc during Solve as additional arguments.

Tested to work with e.g. the "-threads" argument.

Addresses issue #57.

### CODE REVIEW

**Comment by yingzong** [comment]
This failed in some Java test in `GLPK_MIXED_INTEGER_PROGRAMMING`. Likely not be related to this change. What's the process for continuing?

File: ortools/linear_solver/cbc_interface.cc [hunk 1]
```diff
@@ -363,7 +370,7 @@ MPSolver::ResultStatus CBCInterface::Solve(const MPSolverParameters& param) {
   // through callCbc.
   model.setAllowableFractionGap(relative_mip_gap_);
   // NOTE: Trailing space is required to avoid buffer overflow in cbc.
-  int return_status = callCbc("-solve ", model);
+  int return_status = callCbc(solver_specific_parameters_ + " -solve ", model);
```
Comment (Mizux) [inline ortools/linear_solver/cbc_interface.cc]: not sure this space is safe

**Comment by yingzong** [comment]
Fixed the problem. It wasn't the extra space(s), but (I believe) the move-construction performed by `std::operator+(string)` when there is a constant involved (see http://www.cplusplus.com/reference/string/string/operator+/). The moved string ` -solve ` was probably getting freed in multiple threads.

**Comment by Mizux** [comment]
@yingzong I'll revert this commit see #672 for details

**Pull request merged**